File: java/std_nat/src/fan/std/Unsafe.java
//
// Copyright (c) 2009, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   8 Feb 09  Brian Frank  Creation
//
package fan.std;

import fan.sys.FanObj;
import fanx.main.Sys;
import fanx.main.Type;

/**
 * Unsafe
 */
public final class Unsafe
  extends FanObj
{

  public static Unsafe make(Object val) { return new Unsafe(val); }

  public Unsafe(Object val) { this.val = val; }
  
  private static Type type = null;
  public Type typeof() { if (type == null) { type = Sys.findType("std::Unsafe"); } return type;  }

  public Object val() { return val; }
  
  public Object get() { return val; }

  public boolean isImmutable() { return true; }

  private Object val;

}

File: src/std/js/Unsafe.js
//
// Copyright (c) 2009, Brian Frank and Andy Frank
// Licensed under the Academic Free License version 3.0
//
// History:
//   4 Jun 10  Brian Frank  Creation
//

/**
 * Unsafe.
 */
fan.std.Unsafe = fan.sys.Obj.$extend(fan.sys.Obj);

//////////////////////////////////////////////////////////////////////////
// Constructor
//////////////////////////////////////////////////////////////////////////

fan.std.Unsafe.make = function(val)
{
  var self = new fan.std.Unsafe();
  self.m_val = val;
  return self;
}

fan.std.Unsafe.prototype.$ctor = function()
{
}

//////////////////////////////////////////////////////////////////////////
// Methods
//////////////////////////////////////////////////////////////////////////

fan.std.Unsafe.prototype.$typeof = function () {
  return fan.std.Unsafe.$type;
}

fan.std.Unsafe.prototype.val = function() { return this.m_val; }

fan.std.Unsafe.prototype.get = function() { return this.m_val; }

